contracts/connectors/loantoken/FactoryLoanToken.sol

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2017-2023, OokiDao. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0.
+ */
+
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "contracts/connectors/loantoken/LoanToken.sol";
+import "interfaces/ILoanTokenFactory.sol";
+
+contract FactoryLoanToken is LoanToken {
+  address factory;
+
+  constructor() LoanToken(msg.sender, msg.sender) {
+    factory = msg.sender;
+  }
+
+  fallback() external payable override {
+    if (gasleft() <= 2300) {
+      return;
+    }
+
+    address target = _getTarget();
+    bytes memory data = msg.data;
+    require(!ILoanTokenFactory(factory).isPaused(msg.data), "paused");
```

Comment:
why don't we use `msg.sig` like we used in previous version of `PausableGuardian` ?

Reply:
it is converted to msg.sig with LoanTokenFactory but this can be re-worked to reduce gas costs as it is an external call.